lint/parenthesis-checker.go

```diff
@@ -27,21 +27,35 @@ func NewParenthesisChecker(ctx *Context) *ParenthesisChecker {
 // and offsers the way how to do it.
 func (c *ParenthesisChecker) Check(f *ast.File) []Warning {
 	c.warnings = c.warnings[:0]
-	for _, decl := range collectFuncDecls(f) {
-		if decl.Type.Results == nil {
-			continue
-		}
-		for _, res := range decl.Type.Results.List {
-			c.validateResultDecl(res)
+
+	for _, decl := range f.Decls {
+		switch decl := decl.(type) {
+		case *ast.FuncDecl:
+			if decl.Type.Results == nil {
+				continue
+			}
+			for _, res := range decl.Type.Results.List {
+				c.validateExpr(res.Type)
+			}
+		case *ast.GenDecl:
+			for _, spec := range decl.Specs {
+				spec, ok := spec.(*ast.ValueSpec)
```

Comment:
This will panic for specs that are not `var` or `const`.
Add this to your tests and linter will (most likely) panic:
```go
type foo int
```

Reply:
What you may want to do for now is:
```go
if decl.Tok != token.VAR || decl.Tok != token.CONST {
	continue
}
```

Reply:
When you will handle `token.TYPE`, it's still required to skip `token.IMPORT`, as import specs are not interesting for your checker.